README


Global setup:

 Set up git
  git config --global user.name "Emmanuel Donnet"
  git config --global user.email emmanuel@TICanalyste.be
      

Next steps:

  mkdir netOffice--remix-
  cd netOffice--remix-
  git init
  touch README
  git add README
  git commit -m 'first commit'
  git remote add origin git@github.com:TICanalyste/netOffice--remix-.git
  git push -u origin master
      

Existing Git Repo?

  cd existing_git_repo
  git remote add origin git@github.com:TICanalyste/netOffice--remix-.git
  git push -u origin master
      

Importing a Subversion Repo?

  Click here
      

When you're done:

  Continue



docs/changes.txt
CHANGES

2005-06-11 - NetOffice 2.6.0 beta2
+ Minor fixes and touch ups for a new beta release (madbear)

2005-06-09
+ Update JpGraph to latest stable version 1.18 (madbear)

2005-05-31
+ Fixed reports/resultsreport.php display, reported by alpha. (madbear)

2005-05-30
+ Added links on appropriate pages to the iCalendar script (madbear)

2005-05-29
+ Added basic iCalendar (iCal) support for Task, Milestones, and 
  Meetings: calendar/icalendar.php   (madbear)

2005-05-27
+ Implemented fix for "Screen Redraw Issue in block.class.php"
  bug. (madbear)
+ Fixed Multi-task updating notifications. (madbear)
+ Made all date selection fields editable. (madbear)

2005-05-24
+ All Gantt charts now look the same.  The activity bars are color 
  coded to match the status color and the fill color is red for high 
  priority task.  So you can visually see the state of things. (madbear)
+ Phase listings and Views now have opened/closed task(s) and milestone 
  sections (madbear)

2005-05-20
+ Updated overlib from 4.10 to 4.17 (madbear)
+ Updated jscalendar from 0.9.6 to 1.0 (madbear)

2005-01-27
+ ADDED language editor for multiple language support (dylan)
  It is completely backwards compatible, but people should refrain from
  editing the lang_XX files directly from now and use the editor instead -
  please cvs add the global_xx.php files once you have created them.
  Eventually I plan to phase out the lang_XX.php files.
  Developers should use the global text( ) function to get translated
  strings so the user doesn't get blanks if there isn't a translation
  available.

2005-01-23
+ UPDATED jpgraph to 1.17 final (madbear)

2005-01-20
+  [ 964709 ] Security Hole: tightened permissions for these actions (madbear)

2005-01-06
+ ADDED project calendar (luiswang)
+ Fixed bug with sql error in upgrade_db.php (luiswang)
+ Fixed bug with meeting related tables not maintain when meeting's
  project changed (luiswang)
+ Fixed bug with table tasks_time not maintain when task's project
  changed (luiswang)

2005-01-04
+ ADDED Milestone (luiswang)

2004-12-23 - NetOffice 2.6.0 beta1 (happy holidays)
+ Minor touch ups and fixes (tom and scott)

2004-12-17
+ disabled taskNotification to skip not implemented function (tom)
+ new tasks assigned to user by default
+ new tasks open by default
+ project-view: 20 tasks shown (priority as icon only, no id)

2004-12-12
+ Switched away from the older (and deprecated) long predefined array
  variables ($HTTP_*_VARS).  Now we are using the superglobals that 
  were introduced in PHP 4.1.0 (madbear)

2004-12-06
+ ADDED Meetings (luiswang)

2004-12-05
+ Fixed bug with heading toggles not maintaining state (madbear)

2004-12-03:
+ UPDATED Chinese traditional language files (luiswang)
+ UPDATED phpmailer from 1.71 to 1.72 (madbear)
+ UPDATED javascript overlib from 3.51 to 4.10 and relocated to 
  its own sub-directory within the javascript directory (madbear)
+ UPDATED jscalendar from 0.9.5 to 0.9.6 and relocated to its own
  sub-directory within the javascript directory (madbear)
+ UPDATED jpgraph from 1.14 to 1.17b2, which is tagged as the 
  latest stable and the recommended download (madbear)

2004-11-15:
+ block.class-headingForm() (tom)
+ hack for automatically closing headingToggle by closePaletteIcon() (tom)
+ style for forms (tom)
+ pageTitle written by header() (tom)

2004-11-12:
+ adjusted deepblue-theme (CSS and block-class) (tom)
+ implemented new pageHeader with breadcrumb and Msg-Output (tom)
+ removed all local msg-outputs (tom)
+ introduced as global-vars section at library.php (tom)
+ defined constants for link-types (tom)
+ removed buildLink()-function from block-class to clean up code (tom)
+ added a list of possible page sections (tom)

+ Fixed 'profileSession' -> 'profilSession' in 'viewTopic.php' (tom)

+ Fixed bug in editclients file, referencing wrong object (madbear)                     
+ Added 193 more file type icons for linked content displays (madbear)
+ Improved file_info_type function, now all you need to add an icon 
  for a particular file type is to copy the "<ext>.gif" file to 
  "interface/icones/" directory.  Searches for GIF, JPG, and PNG images, 
  (contributed by Cleber P. de Souza)
  

# NetOffice 2.5.3-pl1 / 2003-01-19
+ Removed usage of "mysql_ping", it is PHP >= 4.3 specific. This
  was the cause of the "Blank Screen" bug if you used older 
  PHP versions than 4.3.x (madbear)
+ Changed how classes are loaded, now we use absolute paths
  rather than relative ones, more reliable (madbear)
+ Switched to the LONG Superglobals (HTTP_*_VARS) for greater 
  compatibility with older PHP versions, still only officially
  support PHP >= 4.1 (madbear)
+ Made some configuration changes to jpgraph.php to look a little
  harder for your fonts directory, else error with message if it
  can't find it. Also set some defines to speed it up (madbear)
+ BUG FIX [878657] modified mail notification footer to include 
  correct link to the login page (madbear)
+ BUG FIX [873511] task sorting on viewproject.php page has 
  been corrected (madbear)
+ BUG FIX [803578] Report Results Sorting has been fixed (madbear)


# NetOffice 2.5.3 / 2003-12-24
+ BUG FIX [ 852489 ] typo error in url makes edit project failing (madbear)
+ BUG FIX [ 779377 ] Bookmarks update doesn't work (madbear)
+ BUG FIX [ 823755 ] Sort doesn't work from Home page (madbear)
+ BUG FIX [ 829199 ] updater doesn't detect last digit (madbear)
+ BUG FIX [ 853785 ] wrong owner when entering task hours (madbear)
+ BUG FIX [ 853143 ] SQL Error on inactive project listing (madbear)
+ BUG FIX [ 853207 ] charset in generated emails now uses setCharset from 
  language files (madbear)
+ BUG FIX [ 857710 ] headerfunction() fatal errors (madbear)
+ BUG FIX [ 858347 ] Email notifications settings don't update correctly (madbear)
+ BUG FIX [ 861562 ] PHP Version checking problem when version is greater than 4 (madbear)
+ BUG FIX: Fixed the project listing where if you do the paging it 

notes/viewnote.php
<?php // $Revision: 1.6 $
/* vim: set expandtab ts=4 sw=4 sts=4: */

/**
 * $Id: viewnote.php,v 1.6 2004/12/20 23:45:03 pixtur Exp $
 * 
 * Copyright (c) 2003 by the NetOffice developers
 * 
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 */

$checkSession = true;
require_once("../includes/library.php");

if ($action == "publish") {
    if ($addToSite == "true") {
        $tmpquery1 = "UPDATE " . $tableCollab["notes"] . " SET published='0' WHERE id = '$id'";
        connectSql("$tmpquery1");
        $msg = "addToSite";
    } 
    if ($removeToSite == "true") {
        $tmpquery1 = "UPDATE " . $tableCollab["notes"] . " SET published='1' WHERE id = '$id'";
        connectSql("$tmpquery1");
        $msg = "removeToSite";
    } 
} 


$tmpquery = "WHERE note.id = '$id'";
$noteDetail = new request();
$noteDetail->openNotes($tmpquery);

$tmpquery = "WHERE pro.id = '" . $noteDetail->note_project[0] . "'";
$projectDetail = new request();
$projectDetail->openProjects($tmpquery);

$teamMember = "false";
$tmpquery = "WHERE tea.project = '" . $noteDetail->note_project[0] . "' AND tea.member = '" . $_SESSION['idSession'] . "'";
$memberTest = new request();
$memberTest->openTeams($tmpquery);
$comptMemberTest = count($memberTest->tea_id);
if ($comptMemberTest == "0") {
    $teamMember = "false";
} else {
    $teamMember = "true";
} 

//--- header ---
$breadcrumbs[]=buildLink("../projects/listprojects.php?", $strings["projects"], LINK_INSIDE);
$breadcrumbs[]=buildLink("../projects/viewproject.php?id=" . $projectDetail->pro_id[0], $projectDetail->pro_name[0], LINK_INSIDE);
$breadcrumbs[]=buildLink("../notes/listnotes.php?project=" . $projectDetail->pro_id[0], $strings["notes"], LINK_INSIDE);
$breadcrumbs[]=$noteDetail->note_subject[0];


require_once("../themes/" . THEME . "/header.php");

//--- content ---
$block1 = new block();
$block1->form = "tdD";
$block1->openForm("../notes/viewnote.php#" . $block1->form . "Anchor");
$block1->headingForm($strings["note"] . " : " . $noteDetail->note_subject[0]);
if ($teamMember == "true" && $_SESSION['idSession'] == $noteDetail->note_owner[0]) {
    $block1->openPaletteIcon();
    $block1->paletteIcon(0, "remove", $strings["delete"]);
    // $block1->paletteIcon(1,"export",$strings["export"]);
    if ($sitePublish == "true") {
        $block1->paletteIcon(2, "add_projectsite", $strings["add_project_site"]);
        $block1->paletteIcon(3, "remove_projectsite", $strings["remove_project_site"]);
    }
    $block1->paletteIcon(4, "edit", $strings["edit"]);
    $block1->closePaletteIcon();
}
else {
	$block1->headingForm_close();
}
if ($projectDetail->pro_org_id[0] == "1") {
    $projectDetail->pro_org_name[0] = $strings["none"];
}

$block1->openContent();
$block1->contentTitle($strings["info"]);

$block1->contentRow($strings["project"], buildLink("../projects/viewproject.php?id=" . $projectDetail->pro_id[0], $projectDetail->pro_name[0], LINK_INSIDE));

if ($noteDetail->note_topic[0] != "") {
    $block1->contentRow($strings["topic"], $topicNote[$noteDetail->note_topic[0]]);
}

$block1->contentRow($strings["subject"], $noteDetail->note_subject[0]);
$block1->contentRow($strings["description"], nl2br($noteDetail->note_description[0]));
$block1->contentRow($strings["date"], $noteDetail->note_date[0]);
$block1->contentRow($strings["owner"], buildLink($noteDetail->note_mem_email_work[0], $noteDetail->note_mem_login[0], LINK_MAIL));

$idPublish = $noteDetail->note_published[0];
if ($sitePublish == "true") {
    $block1->contentRow($strings["published"], $statusPublish[$idPublish]);
}

$block1->closeContent();
$block1->headingForm_close();
$block1->closeForm();

if ($teamMember == "true" && $_SESSION['idSession'] == $noteDetail->note_owner[0]) {
    $block1->openPaletteScript();
    $block1->paletteScript(0, "remove", "../notes/deletenotes.php?project=" . $noteDetail->note_project[0] . "&id=" . $noteDetail->note_id[0] . "", "true,true,false", $strings["delete"]);
    // $block1->paletteScript(1,"export","export.php?","true,true,false",$strings["export"]);
    if ($sitePublish == "true") {
        $block1->paletteScript(2, "add_projectsite", "../notes/viewnote.php?addToSite=true&id=" . $noteDetail->note_id[0] . "&action=publish", "true,true,true", $strings["add_project_site"]);
        $block1->paletteScript(3, "remove_projectsite", "../notes/viewnote.php?removeToSite=true&id=" . $noteDetail->note_id[0] . "&action=publish", "true,true,true", $strings["remove_project_site"]);
    }
    $block1->paletteScript(4, "edit", "../notes/editnote.php?project=" . $noteDetail->note_project[0] . "&id=" . $noteDetail->note_id[0] . "", "true,true,false", $strings["edit"]);
    $block1->closePaletteScript("", "");
}

require_once("../themes/" . THEME . "/footer.php");

?>